base/applications/taskmgr/graphctl.h

```diff
@@ -22,93 +22,61 @@
 
 #pragma once
 
-#define MAX_PLOTS 4
-#define MAX_CTRLS 4
+#define NUM_PLOTS    2
+#define PLOT_SHIFT   2
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-#if 0
-
-/* Attributes */
-public:
-  void SetXUnits(const char* string);
-  void SetYUnits(const char* string);
-
-  /* Operations */
-public:
-  BOOL Create(DWORD dwStyle, const RECT& rect, HWND hParentWnd, UINT nID=NULL);
+extern WNDPROC OldGraphCtrlWndProc;
 
-#endif
+INT_PTR CALLBACK GraphCtrl_WndProc(HWND hWnd, UINT message, WPARAM wParam, LPARAM lParam);
 
-typedef struct
+typedef struct _TM_GRAPH_CONTROL
 {
-  int m_nShiftPixels;          /* amount to shift with each new point */
-  int m_nYDecimals;
-
-  char m_strXUnitsString[50];
-  char m_strYUnitsString[50];
-
-  COLORREF m_crBackColor;                 /* background color */
-  COLORREF m_crGridColor;                 /* grid color */
-  COLORREF m_crPlotColor[MAX_PLOTS];      /* data color   */
-
-  double m_dCurrentPosition[MAX_PLOTS];   /* current position */
-  double m_dPreviousPosition[MAX_PLOTS];  /* previous position */
-
-/* those were protected fields */
-  int m_nHalfShiftPixels;
-  int m_nPlotShiftPixels;
-  int m_nGridOffsetPixels;
-  int m_nClientHeight;
-  int m_nClientWidth;
-  int m_nPlotHeight;
-  int m_nPlotWidth;
-  int m_nGridXPixels;
-  int m_nGridYPixels;
-
-  double m_dLowerLimit;        /* lower bounds */
-  double m_dUpperLimit;        /* upper bounds */
-  double m_dRange;
-  double m_dVerticalFactor;
-
-  HWND     m_hWnd;
-  HWND     m_hParentWnd;
-  HDC      m_dcGrid;
-  HDC      m_dcPlot;
-  HBITMAP  m_bitmapOldGrid;
-  HBITMAP  m_bitmapOldPlot;
-  HBITMAP  m_bitmapGrid;
-  HBITMAP  m_bitmapPlot;
-  HBRUSH   m_brushBack;
-  HPEN     m_penPlot[MAX_PLOTS];
-  RECT     m_rectClient;
-  RECT     m_rectPlot;
-} TGraphCtrl;
+    HWND     hParentWnd;
+    HWND     hWnd;
+    HDC      hdcGraph;
+    HBITMAP  hbmGraph;
+    HPEN     hPenGrid;
+    HPEN     hPen0;
+    HPEN     hPen1;
+    HBRUSH   hBrushBack;
+
+    INT      BitmapWidth;
+    INT      BitmapHeight;
+    INT      GridCellWidth;
+    INT      GridCellHeight;
+    INT      CurrShift;
+    INT      Pad;
+
+    PBYTE    PointBuffer;
+    UINT32   NumberOfPoints;
+    UINT32   CurrIndex;
+
+    FLOAT    ftPixelsPerPercent;
+    INT      DrawSecondaryPlot;
```

Comment:
```suggestion
    BOOL     DrawSecondaryPlot;
```

Reply:
what this "improves"? what the size of BOOL? byte? then yeah, "improve" on making misaligned structs. after all, compilers insert padding for you.

Reply:
The size of the win32 type `BOOL` is the same as the size of `INT` which is 4 bytes.
And your structure wouldn't be misaligned because: default padding is 4 bytes (dword-aligned), and your variable is at the end of the structure too.

Reply:
fine, but apart from this time consuming argument, what does it bring? if it's literally the same. it's "fixing" what is not broken. man, this "discussion" has already gotten bigger, than the whole commit. I am already lost in these commit "suggestions" snippets, it was just a few of functions, doing sh1t. now we sank in debates about formatting almost every item over there. I am fed up. if you don't like it, reject. I won't change it for the sake of satisfying your whims. you produced them more in count, than this commit produced changes.

Reply:
`ShowKernelTimes` is already a BOOL, so it makes sense to have this field having the same type.

https://git.reactos.org/?p=reactos.git;a=blob;f=base/applications/taskmgr/taskmgr.h;hb=e055b0623c94bfad785e67c5a913d544e569ec32#l88